director/director.go

```diff
@@ -1269,6 +1269,51 @@ func registerServerAd(engineCtx context.Context, ctx *gin.Context, sType server_
 		adV2.Version = "unknown"
 	}
 
+	if adV2.Downtimes != nil {
+		// Process received server(origin/cache) downtimes and toggle the director db accordingly when necessary
+		currentTime := time.Now().UTC().UnixMilli()
+		bringItDown := false // Flag to indicate if this server is put in downtime in the traversal of
+		sn := adV2.Name
+		// Check if the server is currently in downtime
+		for _, downtime := range adV2.Downtimes {
+			if downtime.StartTime < currentTime && downtime.EndTime > currentTime || downtime.EndTime == 0 {
```

Comment:
```suggestion
			if downtime.StartTime < currentTime && (downtime.EndTime > currentTime || downtime.EndTime == 0) {
```

Order of operations would allow this to proceed regardless, as long as EndTime was 0.